astropy/modeling/functional_models.py

```diff
@@ -2062,3 +2078,87 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
         z = np.sqrt((x_maj / a) ** 2 + (x_min / b) ** 2)
 
         return amplitude * np.exp(-bn * (z ** (1 / n) - 1))
+
+
+class BlackBody1D(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : `~astropy.units.Quantity`
+        Blackbody temperature
+    bolometric_flux : `~astropy.units.Quantity`
+        The bolometric flux of the blackbody (i.e. the integral over the
+        spectral axis)
+    """
+
+    # We parametrize this model with a temperature and a bolometric flux. The
+    # bolometric flux is the integral of the model over the spectral axis. This
+    # is more useful than simply having an amplitude parameter.
+    temperature = Parameter(default=5000 * u.K, min=0)
+    bolometric_flux = Parameter(default=1 * u.erg / u.cm ** 2 / u.s)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be frequencies in Hz.
+    input_units_allow_dimensionless = True
+
+    def evaluate(self, x, temperature, bolometric_flux):
+        """Evaluate the model.
+
+        Parameters
+        ----------
+        x : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Frquency at which to compute the blackbody. If no units are given,
+            this defaults to Hz.
+
+        temperature : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Temperature of the blackbody. If no units are given, this defaults
+            to Kelvin.
+
+        bolometric_flux : float or `~numpy.ndarray` or `~astropy.units.Quantity`
+            Desired integral for the blackbody.
+
+        Returns
+        -------
+        y : number or ndarray
+            Blackbody spectrum. The units are determined from the units of
+            ``bolometric_flux``.
+        """
+
+        from ..constants import sigma_sb
+        from ..analytic_functions.blackbody import blackbody_nu
```

Comment:
If we agree to move ahead with deprecating `analytic_functions`, we should move the actual calculations here. That means, we also have to ensure `blackbody_lambda` can be done somehow. Also have to move over the tests from there.

I cannot seem to  get the equivalent of `test_blackbody_synphot()` to pass using this model. I'll have to investigate more tomorrow...